### PR TITLE
Add wrangler plugin

### PR DESCRIPTION
**Pull request by DaniFoldi**
Hi :wave:,

This PR aims to implement a plugin for Cloudflare Workers, or more specifically their `wrangler` cli. The config file is `wrangler.toml` with json config in experimental status (has been for quite a while), and contains one relevant entry, the main entrypoint of the worker.

I think I managed to add a correctly minimal fixture this time, the new config api made it even simpler compared to the Astro one :tada:, great work!

### CODE REVIEW

File: packages/knip/test/plugins/wrangler.test.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+const cwd = resolve('fixtures/plugins/wrangler');
+
+test('Find dependencies with the wrangler plugin', async () => {
+  const { /* issues, */ counters } = await main({
+    ...baseArguments,
+    cwd,
+  });
+
+  // console.log(issues);
```
Comment (webpro) [inline packages/knip/test/plugins/wrangler.test.ts:16]: Please remove commented out code

File: packages/knip/src/plugins/index.ts [hunk 2]
```diff
@@ -54,3 +54,4 @@ export { default as vue } from './vue/index.js';
 export { default as webpack } from './webpack/index.js';
 export { default as wireit } from './wireit/index.js';
 export { default as yorkie } from './yorkie/index.js';
+export { default as wrangler } from './wrangler/index.js';
```
Comment (webpro) [inline packages/knip/src/plugins/index.ts:57]: Please move it up so it's ordered alphabetically

File: packages/knip/src/ConfigurationValidator.ts [hunk 3]
```diff
@@ -77,6 +77,7 @@ export const pluginSchema = z.union([
 ]);
 
 const pluginsSchema = z.object({
+  wrangler: pluginSchema,
```
Comment (webpro) [inline packages/knip/src/ConfigurationValidator.ts:80]: Please move it down so it's ordered alphabetically

**Review by webpro — APPROVED** [review]
Thanks! Seems like some smooth sailing, just a few minor notes and I'll be happy to merge it

**Comment by webpro** [comment]
Please also make sure the tests pass

File: packages/knip/test/plugins/wrangler.test.ts (hunk 1 above)
Comment (DaniFoldi) [reply packages/knip/test/plugins/wrangler.test.ts]: Fixed in the latest commit

File: packages/knip/src/plugins/index.ts (hunk 2 above)
Comment (DaniFoldi) [reply packages/knip/src/plugins/index.ts]: Fixed in the latest commit

File: packages/knip/src/ConfigurationValidator.ts (hunk 3 above)
Comment (DaniFoldi) [reply packages/knip/src/ConfigurationValidator.ts]: Fixed in the latest commit

**Comment by DaniFoldi** [comment]
Hi,

Thanks for the quick review! I'm actually not sure why the test is failing, at first it didn't fail locally, but now it is so I'm not sure why that was. I'm not sure what the right way to fix the test is, actually, shouldn't `wrangler` being the enabler mark it as used? Or should I add a script to package.json that references it?

**Comment by webpro** [comment]
> shouldn't `wrangler` being the enabler mark it as used? Or should I add a script to package.json that references it?

Usually it's best to have it close to a real-world scenario, so both sounds good :)

**Comment by DaniFoldi** [comment]
Well I gave it a go 😅, now it's complaining about wrangler being unused and unlisted at the same time, I'm quite tired so I'll probably fix it tomorrow:
```
    +   binaries: 1,
    -   binaries: 0,
        classMembers: 0,
    +   dependencies: 1,
    -   dependencies: 0,
    ```

**Comment by webpro** [comment]
Knip checks the `package.json#bin` field of the package in `node_modules`, so I just added that :)

**Comment by webpro** [comment]
:rocket: _This pull request is included in v5.6.0. See [Release 5.6.0](https://github.com/webpro/knip/releases/tag/5.6.0) for release notes._

**Comment by webpro** [comment]
Thanks again, Dániel!

**Comment by DaniFoldi** [comment]
Makes sense, thank you!

**Pull request merged**